lib/rubocop/cop/style/struct_inheritance.rb

```diff
@@ -0,0 +1,40 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for inheritance from Struct.new
+      #
+      # @example
+      #
+      # # bad
+      # class Person < Struct.new(:first_name, :last_name)
+      #
+      # # good
+      # Person = Struct.new(:first_name, :last_name) do
+      class StructInheritance < Cop
+        MSG = "Don't extend a Struct.new - it already is a new class."
```

Comment:
The message sounds a bit strange. You're not extending `Struct.new`, you're extending its result.